src/app/hooks/useTimeout.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2024 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useEffect, useState } from 'react';
+
+export const useTimeout = (msTimeout: number = 30000) => {
+	const [isTimeOut, setIsTimeOut] = useState(false);
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			setIsTimeOut(true);
+		}, msTimeout);
+		return () => clearTimeout(timer);
+	}, []);
+	return isTimeOut;
+};
```

Comment:
The arbitrary 30seconds here is surprising to see. Do we use this default anywhere?

If this is keeping the default value I would add TSDocs to explicitly call it out. 

If I was writing this today, I would not put a default in the function. If we wanted a common timeout value for many instances of this that are using it for a similar operation, then we could reference a global constant that can be modified through config. Use that in each instance instead of setting a default value here.

Reply:
The only change I've made here is move it to a common more descriptive (clear it's a hook) file at a higher level (avoids nasty relative imports)

I'm not clear on what this is doing. This PR is copy/paste from `platform-ui` as we keep saying in our planning.
I'm hoping to clean it up a bit before deployed fully.
At best this appears like it's just making the UI look a little nicer instead of flashing of content.. at worse I think it's an odd workaround for how we're using hooks in places. Requires more work.